tests/unit/s2n_send_test.c

```diff
@@ -54,10 +87,60 @@ int main(int argc, char **argv)
         EXPECT_FAILURE_WITH_ERRNO(s2n_send(conn, test_data, sizeof(test_data), &blocked),
                 S2N_ERR_IO);
         EXPECT_TRUE(s2n_custom_send_fn_called);
+        EXPECT_EQUAL(0, conn->wire_bytes_out);
+
+        /* Cleanup */
+        EXPECT_SUCCESS(s2n_connection_free(conn));
+    }
+
+    /* s2n_send tracks conn->wire_bytes_out on send */
+    {
+        struct s2n_connection *conn;
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+        EXPECT_OK(s2n_connection_set_secrets(conn));
+
+        EXPECT_EQUAL(0, conn->wire_bytes_out);
+
+        EXPECT_SUCCESS(s2n_connection_set_send_cb(conn, s2n_track_sent_bytes_fn));
+        EXPECT_SUCCESS(s2n_connection_set_send_ctx(conn, (void*) conn));
+
+        /* Send test data */
+        uint8_t test_data[] = "hello world";
+        s2n_blocked_status blocked = 0;
+
+        s2n_custom_send_fn_called = false;
+        EXPECT_EQUAL(sizeof(test_data), s2n_send(conn, test_data, sizeof(test_data), &blocked));
+        EXPECT_TRUE(s2n_custom_send_fn_called);
+        EXPECT_EQUAL(sent_bytes, conn->wire_bytes_out);
+        EXPECT_EQUAL(sent_bytes, s2n_connection_get_wire_bytes_out(conn));
+        EXPECT_EQUAL(conn->wire_bytes_out, s2n_connection_get_wire_bytes_out(conn));
 
         /* Cleanup */
         EXPECT_SUCCESS(s2n_connection_free(conn));
     }
 
+    /* s2n_send tracks conn->wire_bytes_out on partial send */
+    {
+        DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT),
+                s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
```

Comment:
The DEFER_CLEANUP allocation should replace the one in EXPECT_NOT_NULL. Instead now you've got two connections. 
```suggestion
        DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT),
                s2n_connection_ptr_free);
        EXPECT_NOT_NULL(conn);
```